network/src/main/java/bisq/network/p2p/services/peergroup/PeerGroup.java

```diff
@@ -65,16 +65,16 @@ public static Config from(com.typesafe.config.Config typesafeConfig) {
     @Getter
     private final Set<Address> seedNodeAddresses;
     private final BanList banList;
-    private final PeerGroupStore peerGroupStore;
+    private final PeerGroupService peerGroupService;
     @Getter
     private final Set<Peer> reportedPeers = new CopyOnWriteArraySet<>();
 
-    public PeerGroup(Node node, Config config, Set<Address> seedNodeAddresses, BanList banList, PeerGroupStore peerGroupStore) {
+    public PeerGroup(Node node, Config config, Set<Address> seedNodeAddresses, BanList banList, PeerGroupService peerGroupService) {
```

Comment:
PeerGroupService creates PeerGroup so it should not have a bi-directional dependency. If needed we can use loose coupling via Interface or handler.
See PeerExchangeService how it uses PersistedPeersHandler

Reply:
Good point, I didn't realize that.